### ISSUE TITLE
`x test src/librustdoc` with download-rustc enabled builds compiler artifacts and fails with dylib error

### ISSUE THREAD

**Issue by camelid** [rust-lang/rust#91071]
It builds compiler artifacts, prints this warning (neither `--stage 2` nor `--stage 3` fix the issue)

```
warning: `download-rustc` does nothing when building stage1 tools; consider using `--stage 2` instead
```

and fails with this dylib error:

```
   Doc-tests rustdoc
dyld[87760]: Library not loaded: @rpath/libtest-42098205ffd4ab92.dylib
  Referenced from: /Users/noahlev/code/rust/r1/build/aarch64-apple-darwin/stage1/bin/rustdoc
  Reason: tried: '/Users/noahlev/code/rust/r1/build/aarch64-apple-darwin/stage1/lib/libtest-42098205ffd4ab92.dylib' (no such file), '/Users/noahlev/code/rust/r1/build/aarch64-apple-darwin/stage1/lib/rustlib/aarch64-apple-darwin/lib/libtest-42098205ffd4ab92.dylib' (no such file), '/Users/noahlev/code/rust/r1/build/aarch64-apple-darwin/stage1/bin/../lib/libtest-42098205ffd4ab92.dylib' (no such file), '/Users/noahlev/code/rust/r1/build/aarch64-apple-darwin/stage1/bin/../lib/libtest-42098205ffd4ab92.dylib' (no such file), '/Users/noahlev/code/rust/r1/build/aarch64-apple-darwin/stage1-tools/release/deps/libtest-42098205ffd4ab92.dylib' (no such file), '/Users/noahlev/code/rust/r1/build/aarch64-apple-darwin/stage1/lib/libtest-42098205ffd4ab92.dylib' (no such file), '/Users/noahlev/lib/libtest-42098205ffd4ab92.dylib' (no such file), '/usr/local/lib/libtest-42098205ffd4ab92.dylib' (no such file), '/usr/lib/libtest-42098205ffd4ab92.dylib' (no such file)
error: test failed, to rerun pass '--doc'
```

The tests pass with download-rustc turned off.

cc @jyn514

**Comment by jyn514** [comment]
>   Referenced from: /Users/noahlev/code/rust/r1/build/aarch64-apple-darwin/stage1/bin/rustdoc

Hmm, this looks weird, I would expect it to be using stage 2 ...

**Comment by jyn514** [comment]
Oh, these are unit tests and not UI tests. I don't know the cause of this error but I'm not surprised it's buggy, I've never tried this.

I think it should be possible in theory? Since rustdoc should link to the downloaded rustc_driver and libtest. If I had to guess the bug is that bootstrap is playing games with rustdoc stages again and hard-coding stage 1 for unit tests.

**Comment by jyn514** [comment]
Oh they're not even unit tests, they're doctests. Yeah that sounds extremely cursed, I would expect it to be a lot of work to fix. That will be in addition to the staging bug though, --no-doc also fails for me.
